goodpeer/goodpeer.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python3
+from collections import OrderedDict
+from pyln.client import Plugin, Millisatoshi
+
+plugin = Plugin()
+
+
+def get_channel_infos(chan):
+    """
+    Filter out what we need from a channel entry in listchannels
+    """
+    return {
+        "scid": chan["short_channel_id"],
+        "ppm_fee": chan["fee_per_millionth"],
+        "base_fee": chan["base_fee_millisatoshi"],
+        "value": chan["amount_msat"],
+    }
+
+
+def get_nodes(plugin):
+    """
+    Get public Lightning network nodes.
+
+    :return: A dict with the node id as key, and a list of channels as values.
+    """
+    peers = {}
+    channels = plugin.rpc.listchannels()
+    for chan in channels["channels"]:
+        if chan["fee_per_millionth"] == chan["base_fee_millisatoshi"] == 0:
+            # They are likely an agreement or the same person, they'd just
+            # distort the medians.
+            continue
+        if not chan["active"]:
+            # It could just be temporary, but we don't want zombies!
+            continue
+        id = chan["source"]
+        if id not in peers.keys():
+            peers[id] = [get_channel_infos(chan)]
+        else:
+            peers[id].append(get_channel_infos(chan))
+    return peers
+
+
+def filter_nodes(nodes):
+    """
+    Filter out nodes with <=15 non-dust channels.
+
+    20$ minimum for a bitcoin at 8000$ (250000000 msats).
+    """
+    filtered_nodes = {}
+    dust_limit = Millisatoshi(250000000)
+    for id, channs in nodes.items():
+        if len(list(filter(lambda c: c["value"] > dust_limit, channs))) > 15:
+            filtered_nodes[id] = channs
+    return filtered_nodes
+
+
+def get_medians(channels):
+    """
+    Return the (median_base_fee, median_fees_ppm) of the given peer.
```

Comment:
Which direction are you considering here? Or does that not matter? I would have expected merchants to look for nodes with a small incoming fee, and end-users for a small outgoing fee.

Reply:
Ah right, I was only considering end-user here !

Reply:
(Considered outgoing fees)

Reply:
I added another optional parameter

Reply:
Actually the `direction` argument doesn't make that much sense, as when scanning nodes I take the fees *this node* sets (`source`), taking the fees its peers set (`destination`) doesn't seem relevant.

> I would have expected merchants to look for nodes with a small incoming fee, and end-users for a small outgoing fee.

Re-thinking it I had outgoing fees in mind in any case, and assumed that if we fund a channel to a peer which always set somewhat small outgoing fees, it would set it to us too. So that we can route through it, and receive through it.